### ISSUE TITLE
Test 'python train.py': when training the wavenet was started, I could get the loss exploded error.

### ISSUE THREAD

**Issue by Yeongtae** [Rayhane-mamah/Tacotron-2#146]
mulaw-quntize is fine. 
When i used 'raw', The problem occurred.
![image](https://user-images.githubusercontent.com/13597229/43987594-d396d3d2-9d5d-11e8-9e0c-6e212463f9d4.png)

Even though I have applied gradient clipping and have reduced learning rate, I could fix it.
@Rayhane-mamah Any idea?

**Comment by Rayhane-mamah** [comment]
Could you send your hparams.py when using "raw" mode please? thanks :)

**Comment by Yeongtae** [comment]
[hparams.zip](https://github.com/Rayhane-mamah/Tacotron-2/files/2280336/hparams.zip)

**Comment by Rayhane-mamah** [comment]
Did you make sure to redo the preprocessing when changing to "raw" and to 2**16 quantize_channels? those changes only occur when doing the wavenet_preprocessing again (or general preprocessing if training with Tacotron)

**Comment by Yeongtae** [comment]
Did you mean that I must redo preprocessing and this part when i change the mode such as 'raw', 'mulaw' and 'mulaw-quntize'?
![image](https://user-images.githubusercontent.com/13597229/43993124-e3e45adc-9dc3-11e8-8a4d-31584bc94ff3.png)

**Comment by Yeongtae** [comment]
Using your advice and 'raw', I have solved it. But I could get - loss, 
![image](https://user-images.githubusercontent.com/13597229/43998863-0b648ec0-9e3b-11e8-9e27-2ea40b1bbe19.png)
![image](https://user-images.githubusercontent.com/13597229/43998877-51825e82-9e3b-11e8-8f32-925ae5d55167.png)
It is going to a larger negative value.

**Comment by Rayhane-mamah** [comment]
Yes that's because it's using gaussian distribution to minimize negative
log probability "-log P(x|x<t;c)". If you prefer to work with usual mixture
of logistics change "output_channels" in wavenet parameters from 2 to 3 *
num_mixture you choose ( I recommend 10)

**Comment by Yeongtae** [comment]
@Rayhane-mamah could you share that Baidu's paper?

**Comment by Rayhane-mamah** [comment]
@Yeongtae sure there you go:
https://arxiv.org/pdf/1807.07281.pdf

**Comment by gloriouskilka** [comment]
BTW, I got loss peak and loss explosion on almost default parameters, `input_type="raw"`, but changed `tacotron_batch_size = 8`. Restarted before that peak, now it passed it ok and training.

**Comment by Ziye-Yang** [comment]
Hi @Rayhane-mamah , I have the wavenet trained for about 5.1k steps and the loss suddenly exploded from 7.48 to 766. This loss explosion stopped the training.  Do you know why?

**Comment by Rayhane-mamah** [comment]
@gloriouskilka and @Ziye-Yang , yes sometimes (at very early stages of training, usually before 7k) loss can explode because those are the steps where the learning rate is either ascending (warmup) or at its peek, normally resuming from checkpoint will go fine as it rarely happens, until I figure out proper ways of having a more stable way to get through this, please simply relaunch training from checkpoint and it will continue just fine. (when we don't use warmup scheme this actually never happens)
